api/middleware/middleware.go

```diff
@@ -92,10 +95,13 @@ func storeDatasets(c *gin.Context, datasets []string) *gin.Context {
 }
 
 // GetDatasets extracts the dataset list from the request context
-var GetDatasets = func(c *gin.Context) []string {
-	datasets := c.GetStringSlice(datasetsKey)
-
-	log.Debugf("returning %v from request context", datasets)
+var GetDatasets = func(c *gin.Context) session.DatasetCache {
```

Comment:
I don't know why it was a `var` but I don't see any reason to have it as such now, change to common `func` ?

Reply:
the `var` allows the function to be easily mocked by replacing the internals in the tests like this https://github.com/neicnordic/sda-download/blob/feature/selectable-middleware/api/sda/sda_test.go#L27-L32

Reply:
Ah, yes, makes sense.